app/uk/gov/hmrc/merchandiseinbaggage/controllers/PreviousDeclarationDetailsController.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggage.controllers
+
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.merchandiseinbaggage.config.AppConfig
+import uk.gov.hmrc.merchandiseinbaggage.connectors.MibConnector
+import uk.gov.hmrc.merchandiseinbaggage.model.api.{Amendment, DeclarationId, Goods}
+import uk.gov.hmrc.merchandiseinbaggage.views.html.PreviousDeclarationDetailsView
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+import uk.gov.hmrc.merchandiseinbaggage.controllers.routes._
+import java.time.LocalDate
+
+@Singleton
+class PreviousDeclarationDetailsController @Inject()(
+  override val controllerComponents: MessagesControllerComponents,
+  actionProvider: DeclarationJourneyActionProvider,
+  mibConnector: MibConnector,
+  view: PreviousDeclarationDetailsView)(implicit ec: ExecutionContext, appConf: AppConfig)
+    extends DeclarationJourneyController {
+
+  val declarationNotFoundMessage = "original declaration was not found"
+
+  val onPageLoad: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
+    // TODO: fix to match what is being provided
+    mibConnector.findDeclaration(request.declarationJourney.declarationId).map {
+      case Some(declaration) =>
+        val now = LocalDate.now
+        val travelDate = declaration.journeyDetails.dateOfTravel
+        val goods = listGoods(declaration.declarationGoods.goods, declaration.amendments)
+        Ok(view(goods, declaration.journeyDetails, declaration.declarationType, withinTimerange(travelDate, now)))
+      case _ =>
+        actionProvider.invalidRequest(s"declaration not found for id:${request.declarationJourney.declarationId.value}")
```

Comment:
i see you are not using the `declarationNotFoundMessage`  you created above ? 
you can remove the // TODO: as well...
also in the `case _` its some technical error and we cannot show the `actionProvider.invalidRequest`  ? because the `declarationId` we have is only after finding a valid declaration for the user entered ref in the previous page